device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/IotHubStatusCode.java

```diff
@@ -3,26 +3,91 @@
 
 package com.microsoft.azure.sdk.iot.device;
 
-import com.microsoft.azure.sdk.iot.device.exceptions.*;
+import com.microsoft.azure.sdk.iot.device.exceptions.IotHubClientException;
+import com.microsoft.azure.sdk.iot.device.transport.IotHubServiceException;
+import com.microsoft.azure.sdk.iot.device.transport.https.exceptions.*;
 
 /**
  * An IoT Hub status code. Included in a message from an IoT Hub to a device.
  */
 public enum IotHubStatusCode
 {
+    /**
+     * The request completed without exception
+     */
     OK,
+
+    /**
+     * The request failed because it had one or more format issues.
+     */
     BAD_FORMAT,
+
+    /**
+     * The request failed because the provided credentials are out of date or incorrect.
+     */
     UNAUTHORIZED,
-    TOO_MANY_DEVICES,
-    HUB_OR_DEVICE_ID_NOT_FOUND,
+
+    /**
+     * The request failed because the quota for such operations has been exceeded. For file upload operations, this
+     * signifies that the maximum number of concurrent file upload operations are already happening. For telemetry operations,
+     * this signifies that the IoT Hub has reached its daily quota for messages ingested.
+     */
+    QUOTA_EXCEEDED,
+
+    /**
+     * The request failed because the resource the request targeted does not exist.
+     */
+    NOT_FOUND,
+
+    /**
+     * The request failed because the request provided an out of date ETag or version number.
+     */
     PRECONDITION_FAILED,
+
+    /**
+     * The request failed because the request payload exceeded IoT Hub's size limits.
+     */
     REQUEST_ENTITY_TOO_LARGE,
+
+    /**
+     * The request was rejected by the service because it is handling too many requests right now.
```

Comment:
I'll clarify this because my use of "it" was a bit too vague here. This is for when the service is handling too many requests, though. So it isn't a client failure